docs/dex-specification.md

```diff
@@ -128,6 +128,8 @@ Token swaps in Dex are a simple way to trade one ERC-20 or KIP7 token for anothe
 
 The [`DexPair`](#dexpair) contract defines the low-level swap function, while the [`DexRouter`](#dexrouter) contract performs the swap operation.
 
+**Warning**: There is a risk of reentrancy when tokens are being swapped. This is due to swap calling safe transfer functions, `safeTransfer` and `safeTransferFrom`.
```

Comment:
Our DEX contracts(DexPair, DexRouter, Farming, StakingFactoryPool, StakingFactoty) don’t use safeTransfer or safeTransferFrom functions. However, the DEXswap token (its `safeTransfer` and `safeTransferFrom` functions) can be used separately by any user (send tokens to other contracts). This can lead to re-entrancy attacks from contracts which are not handled by DEX. The `safeTransfer` and `safeTransferFrom` invoke external call `onKIP7Received` on the recipient address in the _checkOnKIP7Received function  to verify if the recipient address is a KIP7Receiver. This onKIP7Received(recipient) call is the weak place and since it is an external call to an unknown contract, where the recipient contract may define any arbitrary logic to be executed,  it is possible to re-enter functions through usage of `safeTransfer` and `safeTransferFrom`.  Users should check the recipient address thoroughly so it doesn't look suspicious to them, doesn’t have malicious code(unverified contracts), if they want to use these functions.